refac: create traits:TodoStorage, UserInterface

diegocoliveira opened the pull request:
* add trait: TodoStorage, UserInterface
* add impl TodoStorage for Todos
* add impl UserInterface for Terminal
* add impl Display for Todo

diegocoliveira committed:
refac: create traits:TodoStorage, UserInterface
* add trait: TodoStorage, UserInterface
* add impl TodoStorage for Todos
* add impl UserInterface for Terminal
* add impl Display for Todo

diegocoliveira committed:
refac: change methods public to private
* modified: Terminal::write_line for private
* modified: Terminal::title for private

diegocoliveira commented:
Lendo o projeto 7 se eu entendi é para fazer o método done. Esse método eu já fiz, junto com excluir e alterar. Nesse caso aprovado essa PR posso ir direto para o projeto 8?

luisholanda commented on src/cli.rs:
```diff
@@ -0,0 +1,106 @@
+    fn edit(&mut self) -> Result<(), TerminalError> {
+        self.user_interface.list_todo(self.todo_storage.list())?;
+        let id = self.user_interface.select_todo()?;
+        let id = id.parse::<i32>().unwrap_or(0);
```
Fazer o parsing aqui me soa estranho, que tal fazer `select_todo` já retornar o id em si?

diegocoliveira replied:
Verdade... vou ajustar.

luisholanda reviewed (approved):
Apenas uma pequena sugestão, de resto, ficou muito bom!!

diegocoliveira committed:
refac: return fn terminal::select_todo

*before:fn select_todo(&mut self) -> Result<String, TerminalError>;
*after: fn select_todo(&mut self) -> Result<Option<u32>, TerminalError>;
*change id type from i32 to u32
